.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,38 @@
+name: Deploy
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  validate:
+    uses: ./.github/workflows/validate.yaml
+
+  deploy:
+    runs-on: ubuntu-latest
+
+    needs: [ validate ]  # also depends on build for caching build, not repeated to avoid building twice
```

Comment:
It is more clear for me like this:
```suggestion
    needs: [ validate ]  # also depends on `build-and-cache` job for caching build, not repeated to avoid building twice
```